CONTRIBUTING.md

```diff
@@ -345,6 +345,38 @@ with clear instructions on how to install the required libraries.
 Furthermore, if your package requires CGO, it MUST be able to compile and operate in a no op mode
 or report a warning back to the collector with a clear error saying CGO is required to work.
 
+### Breaking changes
+
+Whenever possible, we adhere to semver as our minimum standards. Even before v1, we strive to not break compatibility
+without a good reason. Hence, when a change is known to cause a breaking change, it MUST be clearly marked in the
+changelog, and SHOULD include a line instructing users how to move forward.
+
+We also strive to perform breaking changes in two stages: 
+
+- when we need to remove something, we MUST mark a feature as deprecated in one version, and MAY remove it in the next one
+- when renaming types, functions, or attributes, we MUST create the new name first and MUST deprecate the old one in one
```

Comment:
Is this 3 different steps in 3 versions or 2 steps in 2 versions? Wording is not clear.

Also in some rare case it may not possible, e.g. we have A and B and need to swap the names (I can't think of a more realistic use case, but I suspect there may be). I think SHOULD is more appropriate than MUST here.

Reply:
Two steps in two versions. I changed the text a bit to try to clarify this.

> Also in some rare case it may not possible, e.g. we have A and B and need to swap the names (I can't think of a more realistic use case, but I suspect there may be). I think SHOULD is more appropriate than MUST here.

How about adding a paragraph about handling exceptions to the rules in general?

Reply:
I added a new paragraph about the exceptions.